### PR TITLE
FIX when raven unavailable

### PR DESCRIPTION
**Pull request by bealdav**
Not tested but should works

@pedrobaeza @naglis

### CODE REVIEW

File: sentry/const.py [hunk 1]
```diff
@@ -34,10 +35,10 @@ def split_multiple(string, delimiter=',', strip_chars=None):
 
 DEFAULT_TRANSPORT = 'threaded'
 TRANSPORT_CLASS_MAP = {
-    'requests_synchronous': raven.transport.RequestsHTTPTransport,
-    'requests_threaded': raven.transport.ThreadedRequestsHTTPTransport,
-    'synchronous': raven.transport.HTTPTransport,
-    'threaded': raven.transport.ThreadedHTTPTransport,
+    'requests_synchronous': getattr(raven, 'transport.RequestsHTTPTransport', False),
```
Comment (pedrobaeza) [inline sentry/const.py:38]: getattr is not recursive AFAIK. Better to move this to try section

Comment (bealdav) [reply sentry/const.py:38]: or add raven_import = True in try and False in exception and then add a conditional dict

Comment (pedrobaeza) [reply sentry/const.py:38]: Yes, it's another possibility

**Comment by naglis** [comment]
Hi guys,

sorry for breaking `10.0`, did not really anticipate it.

AFAICT, this will not be enough, as imports from raven are used elsewhere at module level.

Had a [quick go at it](https://github.com/OCA/server-tools/compare/10.0...versada:fix_missing_raven), covering (I hope) all cases where imports from `raven` are used at top-level, although it's not pretty. Will test it more thouroughly tomorrow.

Please let me know if you have better ideas how to solve it (I especially dislike the part of [`SentryHandler` and `SanitizePasswordsProcessor` setting to `object`](https://github.com/versada/server-tools/blob/522c0cf7d5ae4b54d1cbc53cf0c8ed753982a643/sentry/logutils.py#L17-L18) if `raven` can't be imported).

**Comment by bealdav** [comment]
:+1: thanks

**Comment by bealdav** [comment]
Guidelines improvement https://github.com/OCA/maintainer-tools/pull/279/files

**Comment by pedrobaeza** [comment]
Are you going to change this PR according the advise from @naglis?

**Comment by bealdav** [comment]
No @naglis have done this branch https://github.com/OCA/server-tools/compare/10.0...versada:fix_missing_raven and is nice

**Pull request closed without merging**